examples/cfd/stokes_mgn/pi_inference.py

```diff
@@ -0,0 +1,525 @@
+
+        # kinematic viscosity
+        if isinstance(nu, str):
+            nu = Function(nu)(*input_variables)
+        elif isinstance(nu, (float, int)):
+            nu = Number(nu)
+
+        # set equations
+        self.equations = {}
+        self.equations["continuity"] = u.diff(x) + v.diff(y) + w.diff(z)
+        self.equations["momentum_x"] = +p.diff(x) - nu * (
+            u.diff(x).diff(x) + u.diff(y).diff(y) + u.diff(z).diff(z)
+        )
+        self.equations["momentum_y"] = +p.diff(y) - nu * (
+            v.diff(x).diff(x) + v.diff(y).diff(y) + v.diff(z).diff(z)
+        )
+        self.equations["momentum_z"] = +p.diff(z) - nu * (
+            w.diff(x).diff(x) + w.diff(y).diff(y) + w.diff(z).diff(z)
+        )
+
+        if self.dim == 2:
+            self.equations.pop("momentum_z")
+
+
+class DNN(torch.nn.Module):
+    """
+    Custom PyTorch model
+    """
+
+    def __init__(self, layers, fourier_features=64):
+        super(DNN, self).__init__()
+
+        # parameters
+        self.depth = len(layers) - 1
+
+        # Fourier features
+        self.fourier_features = fourier_features
+        self.B = 10 * torch.randn((layers[0], fourier_features)).to(
+            "cuda"
+        )  # Random matrix
+
+        # set up layer order dict
+        self.activation = torch.nn.GELU
+
+        layer_list = list()
+        for i in range(1, self.depth - 1):
+            layer_list.append(
+                ("layer_%d" % i, torch.nn.Linear(layers[i], layers[i + 1]))
+            )
+            layer_list.append(("activation_%d" % i, self.activation()))
+
+        layer_list.append(
+            ("layer_%d" % (self.depth - 1), torch.nn.Linear(layers[-2], layers[-1]))
+        )
+        layerDict = OrderedDict(layer_list)
+
+        # deploy layers
+        self.layers = torch.nn.Sequential(layerDict)
+
+    def forward(self, x):
+        # Add Fourier features
+        x_proj = torch.matmul(x, self.B)
+        x_proj = torch.cat([torch.sin(x_proj), torch.cos(x_proj)], dim=-1)
+
+        # Pass through layers
+        out = self.layers(x_proj)
+        return out
+
+
+class MdlsSymDNN(Arch):
+    """
+    Wrapper model to convert PyTorch model to Modulus-Sym model.
+    """
+
+    def __init__(
+        self,
+        input_keys=[Key("x"), Key("y")],
+        output_keys=[Key("u"), Key("v"), Key("p")],
+        layers=[2, 128, 128, 128, 128, 3],
+        fourier_features=64,
+    ):
+        super(MdlsSymDNN, self).__init__(
+            input_keys=input_keys,
+            output_keys=output_keys,
+        )
+
+        self.mdls_model = DNN(layers, fourier_features)
+
+    def forward(self, dict_tensor: Dict[str, torch.Tensor]):
+        x = self.concat_input(
+            dict_tensor,
+            self.input_key_dict,
+            detach_dict=self.detach_key_dict,
+            dim=-1,
+        )
+        out = self.mdls_model(x)
+        return self.split_output(out, self.output_key_dict, dim=1)
```

Comment:
Add comment here explaining where split_output comes from and why it is needed and how it maps to output_keys

Reply:
Added!